# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/txmgr/TransactionItem.h
# Commit: b9ebe58001d7
# Full Hash: b9ebe58001d787bb0bbad6d39ae8681966d5a77f
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-04-04 09:29:13
# Regressor Bug: 1619914
# File Overlap Count: 1
# Description:
#   Bug 1619914 - part 2: Mark transaction class methods and their caller methods as `MOZ_CAN_RUN_SCRIPT r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D69303
# ==============================================================================

diff -r ce2087ddab5c -r b9ebe58001d7 editor/txmgr/TransactionItem.h
--- a/editor/txmgr/TransactionItem.h	Fri Apr 03 08:30:37 2020 +0000
+++ b/editor/txmgr/TransactionItem.h	Fri Apr 03 08:32:01 2020 +0000
@@ -34,18 +34,24 @@
   }
   nsresult GetChild(size_t aIndex, TransactionItem** aChild);
 
-  nsresult DoTransaction();
-  nsresult UndoTransaction(TransactionManager* aTransactionManager);
-  nsresult RedoTransaction(TransactionManager* aTransactionManager);
+  MOZ_CAN_RUN_SCRIPT nsresult DoTransaction();
+  MOZ_CAN_RUN_SCRIPT nsresult
+  UndoTransaction(TransactionManager* aTransactionManager);
+  MOZ_CAN_RUN_SCRIPT nsresult
+  RedoTransaction(TransactionManager* aTransactionManager);
 
   nsCOMArray<nsISupports>& GetData() { return mData; }
 
  private:
-  nsresult UndoChildren(TransactionManager* aTransactionManager);
-  nsresult RedoChildren(TransactionManager* aTransactionManager);
+  MOZ_CAN_RUN_SCRIPT nsresult
+  UndoChildren(TransactionManager* aTransactionManager);
+  MOZ_CAN_RUN_SCRIPT nsresult
+  RedoChildren(TransactionManager* aTransactionManager);
 
-  nsresult RecoverFromUndoError(TransactionManager* aTransactionManager);
-  nsresult RecoverFromRedoError(TransactionManager* aTransactionManager);
+  MOZ_CAN_RUN_SCRIPT nsresult
+  RecoverFromUndoError(TransactionManager* aTransactionManager);
+  MOZ_CAN_RUN_SCRIPT nsresult
+  RecoverFromRedoError(TransactionManager* aTransactionManager);
 
   size_t NumberOfUndoItems() const;
   size_t NumberOfRedoItems() const;